core/src/main/scala/org/apache/spark/sql/SnappyDDLParser.scala

```diff
@@ -162,6 +162,12 @@ abstract class SnappyDDLParser(session: SnappySession)
   final def WEEK: Rule0 = rule { intervalUnit(Consts.WEEK) }
   final def YEAR: Rule0 = rule { intervalUnit(Consts.YEAR) }
 
+  //cube, rollup, grouping sets
+  final def CUBE = rule { keyword(Consts.CUBE) }
+  final def ROLLUP = rule { keyword(Consts.ROLLUP) }
+  final def GROUPING = rule { keyword(Consts.GROUPING) }
+  final def SETS = rule { keyword(Consts.SETS) }
+
```

Comment:
Can you add the return type explicitly (Rule0) else recent IDEA complains for scalastyle (public methods should declare return types)?